crates/stackable-operator/src/commons/networking.rs

```diff
@@ -39,6 +49,76 @@ impl Deref for Hostname {
     }
 }
 
+/// A validated host (either a [`Hostname`] or IP address) type.
+#[derive(Serialize, Deserialize, Clone, Debug, Eq, PartialEq, Hash, Ord, PartialOrd)]
+#[serde(try_from = "String", into = "String")]
+pub enum Host {
+    IpAddress(IpAddr),
+    Hostname(Hostname),
+}
```

Comment:
note: Currently, our naming is a little wonky. RFC 1123 states that:

> Whenever a user inputs the identity of an Internet host, it SHOULD
> be possible to enter either (1) a host domain name or (2) an IP
> address in dotted-decimal ("#.#.#.#") form.

So a host name is either a valid domain name, which is also enforced by this sentence in the RFC:

> Host software MUST handle host names of up to 63 characters and
> SHOULD handle host names of up to 255 characters.

or an IP address. Therefore, the enum should be defined as:

```suggestion
pub enum Hostname {
    IpAddress(IpAddr),
    Domain(DomainName),
}
```

---

A nice thing we already do correctly is:

> The host SHOULD check the string syntactically for a dotted-decimal
> number before looking it up in the Domain Name System.

Reply:
note: Also on that note, we might want to take a look at the current validation and ensure all regular expressions are nailed down according to the RFCs.

But that is unrelated to this PR, just a general thought I had.